File: notatri.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std ; 
long long solve(vector<long long> arr, long long n){
    long long count = 0, temp_count = 0 , i , j ;
    while(n >= 2){
        temp_count = 0 ;
        i = 0 ; 
        j = n -1 ;
        while(i <j){

            if(arr[i] + arr[j] < arr[n]){
                temp_count += (j - i) ;
                i ++ ;
            }
            else{
                j -- ;
            }
        }
        count += temp_count ;
        n -- ;
        
    }

    return count ; 
}

int main(){
    long long n , i , j , count , temp; 
    while(true){
        vector<long long> arr ;
        cin >> n ; 
        if(n == 0)
            break ;
        for(i = 0 ; i < n ; i++){
            cin >> temp ;
            arr.push_back(temp);
        }
    
        sort(arr.begin(), arr.end());
        cout << solve(arr, n-1)<< endl;
    }
    
return 0 ;
}

File: notatri.py
# IDEA: Divide the sorted array from a point and
# find the number of pairs less than the current 
# value at the index, taking advantage of the sorted array
def sumPairLessThan(v, n, S):
    i = 0 
    j = n 
    count = 0 
    while i < j : 
        # Check the property of triangle
        if v[i] + v[j] < S :
            count += j - i 
            i += 1 
        else:
            j -= 1
    return count

def solve(v, n):
    count  = 0 
    v.sort()
    i = n-1
    while(i >= 2):
        count += sumPairLessThan(v[0:i], i-1 , v[i])
        i -= 1 
    return count

while True:
    n = input()
    if n == 0 :
        exit()
    arr = map(int , raw_input().split())
    print solve(arr, n)
